design/TextSearch.md

```diff
@@ -0,0 +1,49 @@
+# Text Search Design
+
+(aka "BigSearch")
+
+## Key ideas
+
+- Users create text files that provide text data to index and search.
+- Search indexing builds internal resources for faster query results.
+- Each search query leads to zero or more search matches.
+- Each search match includes an internal file position.
+- Files permit queries for data surrounding each match (each internal file position).
+
+### File
+
+- Uniquely-identified resource.
+- All content is text (for now).
+- CRUD API for creating, reading and updating files.
+
+#### File data model
+
+ - Each file is a Sequence of text.
```

Comment:
A text is already a sequence… do you really mean ”Sequence of text” or do you mean “sequence of characters” (= text)?

Reply:
Answered by Claudio